cpp/libs/gst_nmos_sender_plugin/gst_nmos_sender_plugin.c

```diff
@@ -0,0 +1,308 @@
+                                                        "Name of the interface (default: wlp1s0)", "wlp1s0",
+                                                        G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+    g_object_class_install_property(G_OBJECT_CLASS(klass), 3,
+                                    g_param_spec_string("destination-address", "Destination Address",
+                                                        "The address of the destination (default: 127.0.0.1)",
+                                                        "127.0.0.1", G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+    g_object_class_install_property(G_OBJECT_CLASS(klass), 4,
+                                    g_param_spec_string("destination-port", "Destination Port",
+                                                        "Port of the destination (default: 9999)", "9999",
+                                                        G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+    gst_element_class_set_static_metadata(element_class, "NMOS Sender", "Sink/Network",
+                                          "Processes raw video and sends it over RTP and UDP to NMOS client",
+                                          "Luis Ferreira <luis.ferreira@bisect.pt>");
+
+    element_class->change_state = gst_nmossender_change_state;
+}
+
+/* Object initialization */
+static void gst_nmossender_init(GstNmossender* self)
+{
+    self->payloader = gst_element_factory_make("rtpvrawpay", "payloader");
+    self->udpsink   = gst_element_factory_make("udpsink", "udpsink");
+
+    if(!self->payloader || !self->udpsink)
+    {
+        GST_ERROR_OBJECT(self, "Failed to create internal elements: payloader or udpsink");
+        return;
+    }
+
+    // default udpsink properties
+    g_object_set(G_OBJECT(self->udpsink), "host", "127.0.0.1", "port", 9999, NULL);
+
+    // Add elements to the bin
+    gst_bin_add_many(GST_BIN(self), self->payloader, self->udpsink, NULL);
+    gst_element_link(self->payloader, self->udpsink);
+
+    // create and configure the sink pad
+    GstPad* payloader_sinkpad = gst_element_get_static_pad(self->payloader, "sink");
+    GstPad* sink_ghost_pad    = gst_ghost_pad_new("sink", payloader_sinkpad);
+
+    if(!sink_ghost_pad)
+    {
+        GST_ERROR_OBJECT(self, "Failed to create ghost pad for sink");
+        gst_object_unref(payloader_sinkpad);
+        return;
+    }
+
+    // added ghost pad to the element to enable
+    // connectivity between layers of element
+    gst_element_add_pad(GST_ELEMENT(self), sink_ghost_pad);
+
+    gst_pad_set_event_function(sink_ghost_pad, gst_nmossender_sink_event);
+
+    gst_object_unref(payloader_sinkpad);
+}
+
+static gboolean plugin_init(GstPlugin* plugin)
+{
+    return gst_element_register(plugin, "nmossender", GST_RANK_NONE, GST_TYPE_NMOSSENDER);
+}
+
+#define VERSION "0.1"
+#define PACKAGE "gst-plugins-ugly"
```

Comment:
These need to be changed.
`PACKAGE` should be the library name.
`PACKAGE_NAME` should be `AMWA NMOS Sender and Receiver Framework Plugins` 
`GST_PACKAGE_ORIGIN`: `https://www.amwa.tv/`
`GST_PACKAGE_LICENSE`: `Apache-2.0`